docs/adr/0004-only-threadsafe-interfaces.md

```diff
@@ -0,0 +1,156 @@
+# Remove support for non-`Send+Sync` interfaces
+
+* Status: proposed
+* Deciders: mhammond, rfkelly, travis; proposed: jhugman, dmose, janerik
+* Date: 2021-03-31
+
+Technical Story: [Issue 419](https://github.com/mozilla/uniffi-rs/issues/419)
+
+## Context and Problem Statement
+
+[ADR-0003](0003-threadsafe-interfaces.md) introduced support for "thread-safe
+interfaces" - possibly leading to the impression that there is such a thing as
+non-threadsafe interfaces and confusion about exactly what the attribute means.
+
+However, the entire concept of non-threadsafe interfaces is a mismomer -
```

Comment:
Nit: misnomer.

Is it misnamed? or is it a misconception?

Reply:
changed to "misconception"